### PR TITLE
chore: create next version PR by manual trigger

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Creates PR like this by triggering the workflow: https://github.com/DanielMSchmidt/cdktf-provider-danieltestingstuff/pull/103

### CODE REVIEW

File: src/next-version-pr.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            `PR_NUMBER_TO_CLOSE=$(gh pr list | grep "cdktf-next-pr" | awk '{ print $1 }')`,
+            `if [ -z "$PR_NUMBER_TO_CLOSE" ]; then`,
+            `  echo "No PR to close"`,
+            `else`,
+            `  gh pr close $PR_NUMBER_TO_CLOSE`,
+            `fi`,
+          ].join("\n"),
+          env: {
+            GH_TOKEN: githubSecret,
+          },
```
Comment (ansgarm) [inline src/next-version-pr.ts:47]: Can't it update the existing PR instead? What is the reason to close and create a new one? (It feels more noisy that way, but idk may be perfectly fine)

File: src/next-version-pr.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          name: "Install",
+          run: "yarn install",
+        },
+        {
+          name: "Upgrade CDKTF",
+          // Replace 'cdktfVersion: "0.12.0"' with 'cdktfVersion: "<next>"'
+          run: [
+            // We need this to get the full tag of the @next version
+            `yarn upgrade cdktf@next`,
+            `CDKTF_VERSION=$(jq .devDependencies.cdktf -r < package.json)`,
```
Comment (ansgarm) [inline src/next-version-pr.ts:59]: ah, nice!

You could also use `yarn info` which could save you an invocation:
```suggestion
            `CDKTF_VERSION=$(yarn info cdktf --json | jq -r '.data | .["dist-tags"] | .next')`,
```

File: src/next-version-pr.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            `cat .projenrc.js`,
+          ].join("\n"),
+        },
+        {
+          name: "Run projen",
+          run: "yarn run upgrade",
+        },
+        {
+          name: "Regenerate bindings",
+          run: "npx cdktf get",
```
Comment (ansgarm) [inline src/next-version-pr.ts:70]: I think the proper command to update bindings would be `yarn fetch` (it does [a bit more than get](https://github.com/hashicorp/cdktf-provider-aws/blob/e6280ad57d36c3488acab3e53e4b6df6bcd15d21/.projen/tasks.json#L134)).

Comment (ansgarm) [reply src/next-version-pr.ts:70]: The other `upgrade-provider` workflow [also runs `compile` and `docs`](https://github.com/hashicorp/cdktf-provider-aws/blob/e6280ad57d36c3488acab3e53e4b6df6bcd15d21/.github/workflows/provider-upgrade.yml#L24-L25). But I think we don't need them as the purpose of this PR is to possibly fail (which will happen at compile probably) 🤔

File: src/next-version-pr.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        },
+        {
+          name: "Create PR",
+          uses: "peter-evans/create-pull-request@v4",
+          with: {
+            token: githubSecret,
+            branch: "cdktf-next-pr",
+            "branch-suffix": "random",
+            "commit-message": "chore: upgrade CDKTF to next version",
+            title: "chore: upgrade CDKTF to next version",
```
Comment (ansgarm) [inline src/next-version-pr.ts:80]: I'd like to be a bit less ambiguous:
```suggestion
            title: "chore: preview upgrading CDKTF to @next version",
```

File: src/next-version-pr.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          run: "npx cdktf get",
+        },
+        {
+          name: "Create PR",
+          uses: "peter-evans/create-pull-request@v4",
+          with: {
+            token: githubSecret,
+            branch: "cdktf-next-pr",
+            "branch-suffix": "random",
+            "commit-message": "chore: upgrade CDKTF to next version",
```
Comment (ansgarm) [inline src/next-version-pr.ts:79]:
```suggestion
            "commit-message": "chore: upgrade CDKTF to @next preview version",
```

File: src/next-version-pr.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            title: "chore: upgrade CDKTF to next version",
+            "delete-branch": true,
+            draft: true,
```
Comment (ansgarm) [inline src/next-version-pr.ts:82]: Let's make it a bit more clear that this should not be merged (just being draft isn't explicit enough for my taste):
```suggestion
            draft: true,
            body: "DO NOT MERGE! This is an automated PR that tests the pre-built provider generation against preview builds of CDKTF",
```

File: src/next-version-pr.ts (hunk 3 above)
Comment (DanielMSchmidt) [reply src/next-version-pr.ts:69]: I had issues where fetch didn't work, but I can try it again

File: src/next-version-pr.ts (hunk 1 above)
Comment (ansgarm) [reply src/next-version-pr.ts:47]: ah, fair enough 👍

File: src/next-version-pr.ts (hunk 3 above)
Comment (ansgarm) [reply src/next-version-pr.ts:69]: maybe it is good enough to run get, as e.g. copying saving the version to a json file is not necessary as we will never merge the test PR.
I'd still prefer fetch though 😁

**Review by ansgarm — APPROVED** [review]

**Pull request merged**